File: plugins/ruby/Skeleton.rb
require 'java'

java_import 'the.bytecode.club.bytecodeviewer.api.Plugin'
java_import 'the.bytecode.club.bytecodeviewer.api.PluginConsole'
java_import 'java.lang.System'
java_import 'java.util.ArrayList'
java_import 'org.objectweb.asm.tree.ClassNode'

#
# This is a skeleton template for BCV's Ruby Plugin System
#
# @author [Your Name Goes Here]
#

class Skeleton < Plugin
  def execute(classNodeList)
    gui = PluginConsole.new "Skeleton Title"
    gui.setVisible(true)
    gui.appendText("executed skeleton example")
  end
end

File: plugins/python/skeleton.py
from the.bytecode.club.bytecodeviewer.api import Plugin
from the.bytecode.club.bytecodeviewer.api import PluginConsole
from java.lang import System
from java.lang import Boolean
from java.util import ArrayList
from org.objectweb.asm.tree import ClassNode

#
# This is a skeleton template for BCV's Ruby Plugin System
#
# @author [Your Name Goes Here]
#

class skeleton(Plugin):
	def execute(classNodeList, notUsed): #for some reason it requires a second arg
		gui = PluginConsole("Skeleton Title")
		gui.setVisible(Boolean.TRUE)
		gui.appendText("executed skeleton example")